testing/src/main/java/org/oppia/android/testing/junit/OppiaParameterizedTestRunner.kt

```diff
@@ -0,0 +1,303 @@
+package org.oppia.android.testing.junit
+
+import org.junit.runner.Description
+import org.junit.runner.Runner
+import org.junit.runner.manipulation.Filter
+import org.junit.runner.manipulation.Filterable
+import org.junit.runner.manipulation.Sortable
+import org.junit.runner.manipulation.Sorter
+import org.junit.runner.notification.RunNotifier
+import org.junit.runners.Suite
+import java.lang.annotation.Repeatable
+import java.lang.reflect.Field
+import java.lang.reflect.Method
+import kotlin.reflect.KClass
+
+/**
+ * JUnit test runner that enables support for parameterization, that is, running a single test
+ * multiple times with different data values.
+ *
+ * From a testing correctness perspective, this should only be used to test scenarios of behaviors
+ * that are very similar to one another (i.e. only differ in one or two conditions that can be
+ * data-driven), and that have a large number (i.e. >5) conditions to test. For other situations,
+ * use regular explicit tests, instead (since parameterized tests can hurt test maintainability and
+ * readability).
+ *
+ * This runner behaves like AndroidJUnit4 in that it should work in different environments based on
+ * which base runner is configured using [SelectRunnerPlatform] (which automatically pulls in the
+ * necessary Bazel dependencies). However, it will only support the platform(s) selected.
+ *
+ * To introduce parameterized tests, add this runner along with one or more [Parameter]-annotated
+ * fields and one or more [RunParameterized]-annotated methods (where each method should have
+ * multiple [Iteration]s defined to describe each test iteration). Here's a simple example:
+ *
+ * ```kotlin
+ * @RunWith(OppiaParameterizedTestRunner::class)
+ * @SelectRunnerPlatform(ParameterizedRobolectricTestRunner::class)
+ * class ExampleParameterizedTest {
+ *   @Parameter lateinit var parameter: String
+ *
+ *   @Test
+ *   @RunParameterized(
+ *     Iteration("first", "parameter=first value"),
+ *     Iteration("second", "parameter=second value"),
+ *     Iteration("third", "parameter=third value")
+ *   )
+ *   fun testParams_multipleVals_isConsistent() {
+ *     val result = performOperation(parameter)
+ *     assertThat(result).isEqualTo(consistentExpectedValue)
```

Comment:
It may be worth explaining how to pass parameters that are strings as well . ANd to document that only primitives are allowed as parameters. But other than that I think this is great